File: app/src/main/java/com/example/richtext/MVVMActivity.java
package com.example.richtext;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.example.richtext.contract.IMVVMView;
import com.example.richtext.databinding.ActivityMvvmBinding;
import com.example.richtext.model.MVViewModel;
import com.shuyu.textutillib.listener.SpanAtUserCallBack;
import com.shuyu.textutillib.listener.SpanTopicCallBack;
import com.shuyu.textutillib.listener.SpanUrlCallBack;
import com.shuyu.textutillib.model.TopicModel;
import com.shuyu.textutillib.model.UserModel;

public class MVVMActivity extends Activity implements IMVVMView {

    MVViewModel mvViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMvvmBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_mvvm);

        mvViewModel = new MVViewModel(this);
        binding.setViewmodel(mvViewModel);

    }

    @Override
    public SpanTopicCallBack getSpanTopicCallBack() {
        return spanTopicCallBack;
    }

    @Override
    public SpanAtUserCallBack getSpanAtUserCallBack() {
        return spanAtUserCallBack;
    }

    @Override
    public SpanUrlCallBack getSpanUrlCallBack() {
        return spanUrlCallBack;
    }

    @Override
    public Context getContext() {
        return this;
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        mvViewModel.onActivityResult(requestCode, resultCode, data);
    }

    /**
     * 链接回调
     */
    private SpanUrlCallBack spanUrlCallBack = new SpanUrlCallBack() {
        @Override
        public void phone(View view, String phone) {
            Toast.makeText(view.getContext(), phone + " 被点击了", Toast.LENGTH_SHORT).show();
            if (view instanceof TextView) {
                ((TextView) view).setHighlightColor(Color.TRANSPARENT);
            }
        }

        @Override
        public void url(View view, String url) {
            Toast.makeText(view.getContext(), url + " 被点击了", Toast.LENGTH_SHORT).show();
            if (view instanceof TextView) {
                ((TextView) view).setHighlightColor(Color.TRANSPARENT);
            }
        }
    };

    /**
     * at回调
     */
    private SpanAtUserCallBack spanAtUserCallBack = new SpanAtUserCallBack() {
        @Override
        public void onClick(View view, UserModel userModel1) {
            Toast.makeText(view.getContext(), userModel1.getUser_name() + " 被点击了", Toast.LENGTH_SHORT).show();
            if (view instanceof TextView) {
                ((TextView) view).setHighlightColor(Color.TRANSPARENT);
            }
        }
    };

    /**
     * 话题回调
     */
    private SpanTopicCallBack spanTopicCallBack = new SpanTopicCallBack() {
        @Override
        public void onClick(View view, TopicModel topicModel) {
            Toast.makeText(view.getContext(), topicModel.getTopicName() + " 被点击了", Toast.LENGTH_SHORT).show();
            if (view instanceof TextView) {
                ((TextView) view).setHighlightColor(Color.TRANSPARENT);
            }
        }
    };
}

File: app-kotlin/src/main/java/com/example/shuyu/MVVMActivity.kt
package com.example.shuyu

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.databinding.DataBindingUtil
import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.widget.TextView
import android.widget.Toast
import com.example.shuyu.bind.RichEditTextBindingsComponent

import com.example.shuyu.contract.IMVVMView
import com.example.shuyu.databinding.ActivityMvvmBinding
import com.example.shuyu.model.MVViewModel
import com.shuyu.textutillib.listener.SpanAtUserCallBack
import com.shuyu.textutillib.listener.SpanTopicCallBack
import com.shuyu.textutillib.listener.SpanUrlCallBack
import com.shuyu.textutillib.model.TopicModel
import com.shuyu.textutillib.model.UserModel

class MVVMActivity : Activity(), IMVVMView {

    private var mvViewModel: MVViewModel? = null

    override val context: Context
        get() = this

    /**
     * 链接回调
     */
    override val spanUrlCallBack: SpanUrlCallBack = object : SpanUrlCallBack {
        override fun phone(view: View, phone: String) {
            Toast.makeText(view.context, phone + " 被点击了", Toast.LENGTH_SHORT).show()
            if (view is TextView) {
                view.highlightColor = Color.TRANSPARENT
            }
        }

        override fun url(view: View, url: String) {
            Toast.makeText(view.context, url + " 被点击了", Toast.LENGTH_SHORT).show()
            if (view is TextView) {
                view.highlightColor = Color.TRANSPARENT
            }
        }
    }

    /**
     * at回调
     */
    override val spanAtUserCallBack: SpanAtUserCallBack = object : SpanAtUserCallBack {
        override fun onClick(view: View, userModel1: UserModel) {
            Toast.makeText(view.context, userModel1.user_name + " 被点击了", Toast.LENGTH_SHORT).show()
            if (view is TextView) {
                view.highlightColor = Color.TRANSPARENT
            }
        }
    }

    /**
     * 话题回调
     */
    override val spanTopicCallBack: SpanTopicCallBack = object : SpanTopicCallBack {
        override fun onClick(view: View, topicModel: TopicModel) {
            Toast.makeText(view.context, topicModel.topicName + " 被点击了", Toast.LENGTH_SHORT).show()
            if (view is TextView) {
                view.highlightColor = Color.TRANSPARENT
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        DataBindingUtil.setDefaultComponent(RichEditTextBindingsComponent())
        val binding = DataBindingUtil.setContentView<ActivityMvvmBinding>(this, R.layout.activity_mvvm)

        mvViewModel = MVViewModel(this)
        binding.viewmodel = mvViewModel

    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
        super.onActivityResult(requestCode, resultCode, data)
        mvViewModel?.onActivityResult(requestCode, resultCode, data)
    }
}
